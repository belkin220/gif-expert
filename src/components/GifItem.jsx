
import { useState } from "react";
import Swal from 'sweetalert2';
import withReactContent from 'sweetalert2-react-content';
import { capitalize } from "../helpers";

const GifItem = (images) => {
  const MySwal = withReactContent(Swal);
  const [image, setImage] = useState([images]);

  const removeElement = (id) => {
    const newImages = image.filter((img) => img.id !== id);
    setImage(newImages);
    MySwal.fire({
      title: <strong>Good job!</strong>,
      html: <i>{capitalize(images.title) + ' ha sido eliminado con Ã©xito.'}</i>,
      icon: 'success'
    })

  };

  return (
    <>
      {image.map((img) => (
        <div key={img.id} className="col-sm-4 mt-4" >
          <div className="card mb-3">
            <img src={img.url} className="card-img-top" alt={img.title} width={214} height={200} />
            <div className="card-body" style={{ width: '214px', height: '250px' }}>
              <h5 className="card-title">{img.title}</h5>
              <p className="card-text">
                Some quick example text to build on the card title and make up the bulk of the card's content.
              </p>
              <div style={{ position: 'absolute', bottom: '-10px' }}>
                <button className="btn btn-primary" onClick={() => removeElement(img.id)}
                  style={{ marginBottom: '25px' }}>Eliminar
                </button>
              </div>
            </div>
          </div>
        </div>
      ))}
    </>
  );
}

export default GifItem;


