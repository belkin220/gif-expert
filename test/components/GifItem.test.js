import { fireEvent, render, screen } from "@testing-library/react";
import GifItem from "../../src/components/GifItem";

describe('Pruebas en GifItem', () => {

  const title = "Superman";
  const url = "https://marvel.com/superman";
  const error = "Hay props que son requeridas";

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(
    <GifItem title = { title }
             url =  {url } />)
    expect(container).toMatchSnapshot();
  });

  test('Debe mostrar la imagen con el Url y ALT indicados ', () => {
    render(
      <GifItem title = { title }
               url =  { url } />);
      // expect(screen.getByRole('img').src).toBe( url )
      // expect(screen.getByRole('img').alt).toBe( title )

      // Alternativa a lo anterior
      const {src, alt } = screen.getByRole('img');
      expect(src).toBe(url);
      expect(alt).toBe(title);

  });

  test('Debe mostrar el tÂ´tulo en el componente ', () => {
    render(
      <GifItem title = { title }
               url =  { url } />);
               screen.debug()
      expect(screen.getByText(title)).toBeTruthy();

  });
  



})
