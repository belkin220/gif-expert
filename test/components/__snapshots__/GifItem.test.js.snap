// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="col-sm-4"
  >
    <div
      class="card mb-3"
    >
      <img
        alt="Superman"
        class="card-img-top"
        height="200"
        src="https://marvel.com/superman"
        width="214"
      />
      <div
        class="card-body"
        style="width: 214px; height: 250px;"
      >
        <h5
          class="card-title"
        >
          Superman
        </h5>
        <p
          class="card-text"
        >
          Some quick example text to build on the card title and make up the bulk of the card's content.
        </p>
        <button
          class="btn btn-primary"
          style="margin-bottom: 25px;"
        >
          Eliminar
        </button>
      </div>
    </div>
  </div>
</div>
`;
